### PR TITLE
Heatmap example

### PR DESCRIPTION
**Pull request by ryanbaumann**
## Launch Checklist

- [x] briefly describe the changes in this PR
- [x] Lint docs
- [ ] Check that comments for heatmap layer types are clear and accurate 
- [ ] PR Review @mourner @anandthakker 

Closes https://github.com/mapbox/mapbox-gl-js/issues/5420

### PR Changes

- Remove deprecated `contains` expression type from doc generation.
- Add heatmap example for new heatmap layer type
- Remove existing example for heatmaps using legacy styled circle layers
- Update example earthquakes.geojson file with new metadata from USGS website

image 2017-10-07 at 6 34 03 pm

### CODE REVIEW

File: docs/_posts/examples/3400-02-03-heatmap-layer.html [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            "heatmap-intensity": {
+                "stops": [
+                    [0, 1],
+                    [9, 3]
+                ]
+            },
+            //Color ramp for heatmap.  Domain is 0 (low) to 1 (high)
+            "heatmap-color": {
+                "stops": [
+                    [0, "#2166ac"],
```
Comment (mourner) [inline docs/_posts/examples/3400-02-03-heatmap-layer.html]: Let's add a 0-alpha stop a 0. E.g. we can try `#2166ac00`

Comment (ryanbaumann) [reply docs/_posts/examples/3400-02-03-heatmap-layer.html]: @mourner updated the ramp to start with transparency zero.

**Comment by shongololo** [comment]
I'm looking around Mapbox Studio and it doesn't appear that studio supports heatmaps yet? Is this correct? Thanks.

**Comment by ryanbaumann** [comment]
@shongololo correct, heatmaps are only in the latest release of GL JS v0.41.0 so far.  More to come!

**Pull request merged**